### PR TITLE
doc: Development process enhancements

### PR DESCRIPTION
**Pull request by nashif**
- doc: remove outdated and wrong merge-window description
- doc: development process: feature tracking

### CODE REVIEW

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: So we limit the usage of "feature request" and "feature" only to "mayor" features?
Where "mayor" means something of enough caliber to require being added to the roadmap and so forth?

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: You want to use "priority" labels for features too?
(Today we do not do so in any consistent way, but it would make sense)

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: > detailed architectural design

ambitious.. :)

**Review by aescolar — CHANGES_REQUESTED** [review]
2 minors + couple of questions

File: doc/development_process/issues.rst
Comment (dbkinder) [reply doc/development_process/issues.rst]: yep, change ``not a`` to just ``not``

File: doc/development_process/issues.rst
Comment (dbkinder) [reply doc/development_process/issues.rst]: Yep, add some commas here, after each ``yet``

File: doc/development_process/issues.rst
Comment (dbkinder) [reply doc/development_process/issues.rst]: change to ``and an owner``.

File: doc/development_process/issues.rst
Comment (dbkinder) [reply doc/development_process/issues.rst]: From the description below, I read there's a process to get from a Feature Request to a Feature, and in that process, an owner is assigned and a design is documented...

File: doc/development_process/issues.rst
Comment (dbkinder) [reply doc/development_process/issues.rst]: I'd agree.  We shouldn't stop someone requesting a feature because they need it for their project, but aren't capable of implementing it themselves.  How about:
```
This is the formal way for asking for a new feature in
Zephyr and indicating its importance to the project.  Often,
the requester will have a readiness and willingness to drive implementation of
the feature in an upcoming release, and should assign the request to themselves.
If not though, an owner will be assigned after evaluation by the TSC. 
A feature request can also have a companion RFC with more details on the feature and a
proposed design or implementation.
```

File: doc/development_process/issues.rst
Comment (dbkinder) [inline doc/development_process/issues.rst]:
```suggestion
Items labeled as ``features`` are short an long term roadmap items that shall
```

**Review by aescolar — COMMENTED** [review]
By the way, this topic has a significant overlap with the old content from doc/development_process/proposals.rst

File: doc/development_process/issues.rst
Comment (dbkinder) [inline doc/development_process/issues.rst]:
```suggestion
community.  You should contact an assigned owner if you'd like to discuss or contribute
to that feature's implementation 
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: "must go through an RFC process."

What would this RFC process be?
A thought: Today, many features are added by the maintainers of an area with only some internal discussion or not even. And for anything non controversial, which just adds new functionality and does not break anything, it would seem perfectly ok.

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: When you say that something labeled as feature, it must be a roadmap item, you mean that they must be listed in the GH roadmap project directly? or can they be indirectly linked? (thru a "master" issue)

File: doc/development_process/issues.rst
Comment (aescolar) [inline doc/development_process/issues.rst]: Sidecomment: Maybe not in this paragraph, but it would be nice to mention somewhere that this is the TSC responsibility, and that any new features to be added to the roadmap should be brought to the TSC.

**Comment by nashif** [comment]
> By the way, this topic has a significant overlap with the old content from doc/development_process/proposals.rst

yeah, good point. I will consolidate all of that now.

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks for doing this, Anas.
A few requests for improvements below

File: doc/development_process/proposals.rst
Comment (dbkinder) [reply doc/development_process/proposals.rst]: I read this as, an owner (if not the requester) is required for a **feature request** to transition to a **feature** -- no owner would mean no transition. It says something like this later in the doc.

**Review by dbkinder — APPROVED** [review]
LGTM, modulo @aescolar's comments

File: doc/development_process/proposals.rst
Comment (aescolar) [reply doc/development_process/proposals.rst]: @dbkinder : Ok, from the context of the paragraph I did not understand it as referring to a feature request that was accepted to become a feature. The paragraph seemed a bit disconnected from the overall text around.

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: typo: recruit-e-

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: typo: and and

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: extra commas (?): Some changes+,+ though+,+ are (?)

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: typo: when +the+ project (?)

**Review by aescolar — CHANGES_REQUESTED** [review]
A few minor typos. @dbkinder : Note a few new paragraphs have been added (your typo detecting eye may come in handy)

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: This is just a question (which does not necessarily need a change, but needs a reponse :) ): Will "feature requests" be allowed priority? or only "features"?, if so, who sets the priority to a "feature request" and what would the meaning be? (the priority requester is asking the project give it?)

(if these labels only apply to "features" maybe this labels should be moved to be below the next paragraph?)

File: doc/development_process/proposals.rst
Comment (aescolar) [inline doc/development_process/proposals.rst]: typo: nedd => need

**Review by aescolar — CHANGES_REQUESTED** [review]
If it wasn't for the typo I would have already removed the -1

File: doc/development_process/proposals.rst
Comment (nashif) [reply doc/development_process/proposals.rst]: i think it should apply to both and helps with triaging process because a request will become a release feature at some point, so setting the priority on the request keeps track of that.

**Review by aescolar — APPROVED** [review]
Now we just need to start working in this way ;)

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
